JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreens.kt

```diff
@@ -263,66 +254,52 @@ private fun HomeScreenWithList(
         modifier: Modifier
     ) -> Unit
 ) {
-    Row(modifier.fillMaxSize()) {
-        if (showNavRail) {
-            AppNavRail(
-                currentRoute = JetnewsDestinations.HOME_ROUTE,
-                navigateToHome = { /* Do nothing */ },
-                navigateToInterests = navigateToInterests,
-                modifier = Modifier.navigationBarsPadding(
-                    bottom = false,
-                    end = false
+    Scaffold(
+        scaffoldState = scaffoldState,
+        snackbarHost = { JetnewsSnackbarHost(hostState = it) },
+        topBar = {
+            if (showTopAppBar) {
+                HomeTopAppBar(
```

Comment:
When `showTopAppBar` is true, the `HomeTopAppBar` gets the navigation insets applied twice: once from the `homeScreenPadding`, and once from the internal `InsetAwareTopAppBar`:

![device-2021-09-17-140140](https://user-images.githubusercontent.com/4217560/133853333-bba504a7-1ba3-4504-855b-28d4fb289c54.png)